### PR TITLE
feat: add verified option to map dropdown

### LINKED ISSUE #1197: Add verified filter to the map
## Component infos
We need a filter on the map to filter our verified workspace. It would work the same as the existing dropdown filter`(workspace, extrusion, sheetpress, community point etc)` 
- When selected it will show verified workspaces. 

### Assets
The Verified user data is stored in the V3_users>username>badges>verified
The icon can be found [here](https://github.com/ONEARMY/community-platform/blob/master/src/assets/icons/icon-verified-badge.svg) And it uses our aggregation
### Page related

Will be used in :https://community.preciousplastic.com/map

### Mockup
map-dropdown-verified-icon

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [X] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Closes #1197

### CODE REVIEW

**Comment by thisislawatts** [comment]
@davehakkens at the moment the filters are cumulative so if you select `workspace` and `member` you will get all map pins which match either of those criteria. Do we want to preserve the same behaviour for the `verifed` filter? 

I was thinking that a user may want to see _only_ `members` who are also `verified`. If we want to take that approach we may want to adjust the visual styling so that it looks different because it behaves differently.

**Comment by thisislawatts** [comment]
Yes I have currently implemented it so that it will match `verified` AND type, but I wasn't sure if this was the intended behaviour.

**Comment by davehakkens** [comment]
yeh its awesome! Very happy to see this one being developed :)

**Review by davehakkens — APPROVED** [review]

File: src/stores/Maps/filter.ts [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
-  // filter pins to include matched pin type or subtype
-  // excluding items which have been marked as deleted=true
-  return mapPins.filter(
-    (pin) =>
-      !pin._deleted &&
-      (pin.subType
-        ? filters.includes(pin.subType)
-        : filters.includes(pin.type)),
-  )
+  const filterList = new Set(filters)
```
Comment (chrismclarke) [inline src/stores/Maps/filter.ts:9]: ooh, fancy! will get a nice performance boost if we ever expand our list of filters into the thousands too :D

File: src/pages/Maps/Content/Controls/GroupingFilterMobile.tsx [hunk 2]
```diff
@@ -50,69 +43,57 @@ class GroupingFilterMobile extends Component<IProps, IState> {
 
   handleChange(item: string) {
     if (this.props.onChange) {
+      console.log(`handleChange`, { item })
```
Comment (chrismclarke) [inline src/pages/Maps/Content/Controls/GroupingFilterMobile.tsx:46]: Think this debug log may have been left hanging, no big deal but might be good to remove next time working on the code

File: src/pages/Maps/Content/Controls/transformAvailableFiltersToGroups.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+const ICON_SIZE = 30
+
+function asOptions(mapStore, items: Array<IMapGrouping>): FilterGroupOption[] {
+  return (items || [])
+    .filter((item) => {
+      return !item.hidden
+    })
+    .map((item) => {
+      const filterType =
+        (item.type as string) === 'verified'
```
Comment (chrismclarke) [inline src/pages/Maps/Content/Controls/transformAvailableFiltersToGroups.tsx:18]: the use of double ternary makes this a bit harder to read, and the order of checking first for verified type, then subtype then back to type again is a bit confusing (assume a result of confusing code elsewhere and/or just trying to get a quick fix working). If you have time to add a comment to explain briefly the behaviour that would be handy

**Comment by chrismclarke** [comment]
Really nice to have this in, thanks for the addition @thisislawatts!

**Pull request merged**